lib/ansible/modules/windows/win_pagefile.ps1

```diff
@@ -114,11 +108,11 @@ if ($state -eq "absent") {
     # Set pagefile
     try {
         if ((Get-Pagefile $fullPath) -eq $null) {
-            $pagefile = Set-WmiInstance -Class Win32_PageFileSetting -Arguments @{name = $fullPath; InitialSize = 0; MaximumSize = 0}
-            if (!$systemManaged) {
-                $pagefile.InitialSize = $initialSize
-                $pagefile.MaximumSize = $maximumSize
-                if (-not $check_mode) {
+            if (-not $check_mode) {
+                $pagefile = Set-WmiInstance -Class Win32_PageFileSetting -Arguments @{name = $fullPath; InitialSize = 0; MaximumSize = 0}
```

Comment:
Forgive me for being pedantic, but here again I would prefer a `-WhatIf:$check_mode` for Set-WmiInstance.
Obviously the rest of the block does still need the `if (-not $check_mode)`.

The reason why we prefer the `-WhatIf` functionality is that it is likely that the function also does more than just skip it based on the boolean. We expect that where possible the function also verifies other conditions within the call.